### PR TITLE
binaryen 32 (new formula)

### PR DESCRIPTION
**Pull request by alexreg**

### CODE REVIEW

**Comment by fxcoudert** [comment]
Audit fails with:

```
binaryen:
  * Non-executables were installed to "/usr/local/opt/binaryen/bin"
    The offending files are:
      /usr/local/opt/binaryen/bin/binaryen.js
      /usr/local/opt/binaryen/bin/wasm.js
```

**Comment by alexreg** [comment]
Okay. Firstly, is that really a problem? I reported it as an issue, but it seems fix is not coming too soon. Secondly, I wouldn't know how to fix it with a patch. Maybe you would know how to do this?

**Comment by bfontaine** [comment]
Linking things here for a follow-up:
- the issue report: https://github.com/WebAssembly/binaryen/issues/951
- upstream’s PR: https://github.com/WebAssembly/binaryen/pull/958

Once upstream merges the PR, we’ll be able to add a patch with their commit. Unless they release a new version of course, in which case we’d only need to bump the version here.

**Comment by joshuata** [comment]
It looks like the pull request landed in master and issue is now closed.

**Comment by bfontaine** [comment]
Great! We can use a link to the upstream commit. You can get a patch from any commit on GitHub by appending `.patch` to its URL: https://github.com/WebAssembly/binaryen/commit/0f3102d13617e70b598e7d82f12684d8fa01969b.patch

See our docs about adding a patch to a formula: http://docs.brew.sh/Formula-Cookbook.html#patches
It’ll look like this:
```ruby
# Add a comment above the patch block to explain why the patch is necessary
# (not to install stuff under bin/)
patch do
  url "https://github.com/WebAssembly/binaryen/commit/0f3102d13617e70b598e7d82f12684d8fa01969b.patch"
  sha256 "..."
end
```

**Comment by joshuata** [comment]
Unfortunately I can't edit the PR, so we will have to wait for @alexreg to update it.

**Comment by alexreg** [comment]
Sorry for the delay; I was away on holiday for a couple of weeks. Thanks for the suggestions, @bfontaine, @joshuata. I've made the required updates, and it should be all ready to merge now!

**Comment by joshuata** [comment]
This looks like it is ready to merge. Is there anything left to do?

**Comment by joshuata** [comment]
@alexreg Could you update the pull request to version 32? That should cause Jenkins to re-run the tests.

File: Formula/binaryen.rb [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    sha256 "0182bc6310219b7993553c5155661db780c75b3c1372552bd58f364c594324ad"
+  end
+
+  def install
+    ENV.cxx11
+
+    system "cmake", ".", *std_cmake_args
+    system "make", "install"
+
+    prefix.install "test/"
```
Comment (DomT4) [inline Formula/binaryen.rb]: `pkgshare` is probably a better fit here FWIW.

Comment (alexreg) [inline Formula/binaryen.rb]: Thanks, I've updated that too.

**Comment by alexreg** [comment]
@joshuata Gosh, they're making new releases faster than I'm updating the formula! Anyway, that's all done now. And the new release has obviated the need for manually including the patch, so all the better.

**Comment by joshuata** [comment]
This needs #12907 in order to work completely since Binaryen 32 depends on a more recent version of emscripten (1.37.10). This is only a problem at runtime, however, and should not influence this Formula being merged.

**Comment by joshuata** [comment]
@bfontaine @DomT4 Could we get this merged?

**Comment by DomT4** [comment]
I'm just a contributor these days; @ilovezfs or @bfontaine will be the final yay/nay here.

**Comment by fxcoudert** [comment]
Merged, thanks @alexreg for the pull request and everyone who commented!

**Pull request closed without merging**